hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/TensorFlowJob.groovy

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright 2018 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.gradle.hadoopdsl.job;
+
+import com.linkedin.gradle.hadoopdsl.HadoopDslMethod;
+
+/**
+ * Job class for type=TensorFlowJob jobs.
+ * <p>
+ * In the DSL, a TensorFlowJob can be specified with:
+ * <pre>
+ *   tensorFlowJob('jobName') {
+ *     amMemory 2048
+ *     amCores 1
+ *     psMemory 2048
+ *     psCores 1
+ *     workerMemory 8192
+ *     workerCores 1
+ *     numPs 2
+ *     numWorkers 4
+ *     archive 'tensorflow-starter-kit-1.4.16-SNAPSHOT-azkaban.zip'
+ *     jar 'tensorflow-on-yarn-0.0.1.jar'
+ *     taskCommand ''
+ *     set workerEnv: [
+ *       'ENV1': 'val1',
+ *       'ENV2': 'val2'
+ *     ]
+ *   }
+ * </pre>
+ */
+class TensorFlowJob extends HadoopJavaProcessJob {
+  int amMemory;
+  int amCores;
+  int psMemory;
+  int psCores;
+  int workerMemory;
+  int workerCores;
+  int numPs;
+  int numWorkers;
+  String archive;
+  String jar;
+  String taskCommand;
+  Map<String, Object> workerEnv;
+
+  /**
+   * Constructor for a TensorFlowJob.
+   *
+   * @param jobName The job name
+   */
+  TensorFlowJob(String jobName) {
+    super(jobName);
+    workerEnv = new HashMap<>();
+    setJobProperty("type", "TensorFlowJob");
+  }
+
+  /**
+   * Clones the job.
+   *
+   * @return The cloned job
+   */
+  @Override
+  TensorFlowJob clone() {
+    return clone(new TensorFlowJob(name));
+  }
+
+  /**
+   * Helper method to set the properties on a cloned job.
+   *
+   * @param cloneJob The job being cloned
+   * @return The cloned job
+   */
+  TensorFlowJob clone(TensorFlowJob cloneJob) {
+    cloneJob.amMemory = amMemory;
+    cloneJob.amCores = amCores;
+    cloneJob.psMemory = psMemory;
+    cloneJob.psCores = psCores;
+    cloneJob.workerMemory = workerMemory;
+    cloneJob.workerCores = workerCores;
+    cloneJob.numPs = numPs;
+    cloneJob.numWorkers = numWorkers;
+    cloneJob.archive = archive;
+    cloneJob.jar = jar;
+    cloneJob.taskCommand = taskCommand;
+    cloneJob.workerEnv.putAll(workerEnv);
+    return ((TensorFlowJob)super.clone(cloneJob));
+  }
+
+  /**
+   * DSL method to set the am memory.
```

Comment:
As the plugin now is capable of showing the method's docs in IDE, can you please verbose in all the Groovy Docs?
am -> Application Master(AM)

As an extension it would be helpful for users to know the following:
1) Can you briefly describe what the parameter is?
2) what can be set by default?
3) what is the implication of modifying the parameter?

Reply:
Added some more details to the method docs.